### PR TITLE
Allow to pass htmldict option to the jsviewer writer

### PR DESCRIPTION
**Pull request by saimn**
This is needed to specify that a column should not be escaped, as with the HTML writer.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ env:
         - NUMPY_VERSION=stable
         - MAIN_CMD='python setup.py'
         - CONDA_DEPENDENCIES='Cython jinja2'
-        - CONDA_ALL_DEPENDENCIES='Cython jinja2 scipy h5py matplotlib pyyaml scikit-image pandas pytz beautifulsoup4 ipython mpmath'
+        - CONDA_ALL_DEPENDENCIES='Cython jinja2 scipy h5py matplotlib pyyaml scikit-image pandas pytz beautifulsoup4 ipython mpmath bleach'
```
Comment (saimn) [inline .travis.yml]: It seems bleach was not installed on Travis, so the other tests depending on it (in `test_html.py`) were not run.

Comment (bsipocz) [reply .travis.yml]: Thanks @saimn! This is a very similar issue to #6538, so could you also add a note about this optional dependency in ``docs/install.rst``?

Comment (MSeifert04) [reply .travis.yml]: Preferable at a position that won't produce a conflict if any of the PRs is merged? 😄

Comment (saimn) [reply .travis.yml]: Good point, I have added the note, before `mock` to avoid conflicting with #6538 :wink:

**Comment by taldcroft** [comment]
One minor stylistic nit.  The use of `htmldict_` as a variable name made it harder for me to read and it just took a bit longer to get clear in my head the relationship between that and `htmldict`.  Somehow I think of a variable ending with `_` as a temp or throwaway value, but in this case it is the final one that gets used.

Once we get to Py3.5 then you could do the nice one-liner:
```
htmldict = {**htmldict_default, **(htmldict or {})}
```

**Comment by bsipocz** [comment]
> Once we get to Py3.5 then you could do the nice one-liner:

if it helps, just write this PR straight away in python3.5+, and we could just hold off merging a few more days.

**Comment by saimn** [comment]
>  Shouldn't this addition filter up to some API definition at the top or at least some narrative docs? I know we have a problem with documentation of Table.read/write parameters in general.

Yes, that's indeed a more general issue, the read/write parameters are not documented (#1011). 

> Should this new htmldict be visible in show_in_browser?

No, `show_in_browser` doesn't use the HTML writer as it doesn't write an HTML page to disk.

About `htmldict_`, I agree it's confusing, I was too lazy to find another name. But I'm not sure your one-liner improves readability :wink:, I could simply rename `htmldict_` to `html_options` or something similar ?

**Comment by mwcraig** [comment]
Question: is this related to https://github.com/astropy/astropy/pull/5311?

**Comment by saimn** [comment]
@mwcraig - Yes indeed, I missed that one. It is basically the same, though the implementation differ a bit.

**Comment by bsipocz** [comment]
cancelled osx build

**Comment by MSeifert04** [comment]
@bsipocz No time for the OSX travis-test discussion yet? 😄

**Comment by bsipocz** [comment]
haven't got there yet, I plan to have a proposal PR with the stages. However now we have a simplified matrix, so total build time dropped significantly.

**Comment by saimn** [comment]
@taldcroft - Agreed about the need to document these options, currently it's really hidden. I have added a paragraph with a short description of the parameters, is it good enough ?

jsviewer

**Comment by taldcroft** [comment]
@saimn - docs look great.  I'm on slow internet now, if you can confirm the Travis fails are unrelated I can merge.

**Comment by bsipocz** [comment]
Travis actually passed on this with the exception of the cancelled osx. However now there is a conflict.

**Comment by saimn** [comment]
Rebased and all builds are green :)

**Pull request merged**